### ISSUE TITLE
Error Code C1047 when building using Visual Studio 2015

### ISSUE THREAD

**Issue by Plut0nio** [openitg/openitg#109]
I am trying to build OpenITG for Windows XP (32-bit) using Visual Studio 2015 on Windows 10 according to the instructions in README.md and I keep getting the following error:

Severity	Code	Description	Project	File	Line
Error	C1047	The object or library file 'C:\Games\OpenITG\src\/build-vs2015/OpenITG/Release-SSE2/ACIO.obj' was created with an older compiler than other objects; rebuild old objects and libraries	OpenITG	C:\Games\OpenITG\src\LINK	1

I have cleaned and rebuilt the solution several times and I'm still getting the error. Is there a step that I'm missing that will force ACIO to be recompiled? Adding the /FS flag to Additional Options under "C/C++ -> Command Line" didn't work either.

I ran the Beta 2 installer on the XP PC. Should I just install the patch.zip and replace the binaries instead?

**Comment by chrispable** [comment]
Did you set all projects in the solution to use the same tool set? You have to set more than just the openitg project to use the tool set of your choosing. Make sure you use the _xp version of the tools.

In addition, I highly recommend against using vs2015 for XP if you can avoid it. The new layout of the crt in 2015 does not like xp embedded if that's the route your going.

**Comment by Plut0nio** [comment]
@chrispable They all appear to be using the same Platform Toolset (Visual Studio 2015 - Windows XP (v140_xp)). Unfortunately I uninstalled Visual Studio 2012 a while ago when I upgraded to 2015. The OS of the PC I'm building for in question is Windows XP Pro SP3.

**Comment by chrispable** [comment]
The problem isn't with acio. It just happens to be the first file in the
list of files being used. Obj files are also generated by YOUR compiler,
they aren't checked into the source. This probably means your compiler is
not working correctly. It sounds like you are compiling with an older tool
chain and linking with a newer one. Perhaps there's a remnant of vs2012 on
your system? I've tested again from a fresh checkout on 2015, 2013 and 2010
on a clean virtual machine between all of them this morning and had no
issue. Maybe try a VM to isolate a problem potentially with your visual
studio install.

**Comment by Plut0nio** [comment]
If I can't fix this by simply uninstalling 2015 and either installing it again or installing 2012, 2013, or 2010 instead, then I'll definitely look into using a VM. However, it might take a while because I've never used a VM before and I don't want to mess anything up. Should I use 10, 7, or even XP in my VM to build?

**Comment by chrispable** [comment]
I used a Windows 10 VM. Microsoft offers free vms for testing internet
explorer you can hijack and use as a one time quick hit presetup solution.
You'll obviously need to install visual studio and the requesite sdks into
it.

**Comment by Plut0nio** [comment]
That might just be outside of my level of expertise. However, it looks like Microsoft has a free VM for Windows 10 Enterprise that lasts for two weeks. It includes Visual Studio 2017 instead of 2015, though (only the paid Professional version has 2015). I might ask Stack Overflow about this. And by the way, my compiler (cl.exe) and linker (link.exe) versions are 19.00.23026 (for x86) and 14.00.23026.0, respectively:

![vs2015_cl_and_link_versions](https://user-images.githubusercontent.com/24761328/27849044-f11b929a-6117-11e7-8bb1-1605b43a7e4f.PNG)

Thanks anyway!

**Comment by chrispable** [comment]
Those version numbers are correct but when you launch the compiler tool
chain it adds tools to the path and invokes that way. That's why there's a
special command prompt that sets the variables per environment.

Use one of these and do as I said earlier:

https://developer.microsoft.com/en-us/microsoft-edge/tools/vms/

You get 3 months with it. They recommend setting a snapshot once you get it
all good and you can keep it in perpetuity.

**Comment by Plut0nio** [comment]
Ok, I just followed the README.md tutorial using Visual Studio Community 2013 with Update 5 in a Windows 7 VM and I got a bunch of C4005 warnings along with a fatal error C1083 saying that it can't open/find the file "d3dx9tex.h." It ends up with 4 successful builds and 1 failure. I thought a DirectX SDK was already included with VS2013. Do I have to install a legacy DirectX SDK and use that instead? I'm using the v120_xp Platform Toolset to (re)build the solution.

**Comment by chrispable** [comment]
Yes you need to install the dx9 sdk

On Jul 7, 2017 12:26 AM, "Plut0nio" <notifications@github.com> wrote:

Ok, I just followed the README.md tutorial using Visual Studio Community
2013 with Update 5 in a Windows 7 VM and I got a bunch of C4005 warnings
along with a fatal error C1083 saying that it can't open/find the file
"d3dx9tex.h." It ends up with 4 successful builds and 1 failure. I thought
a DirectX SDK was already included with VS2013. Do I have to install a
legacy DirectX SDK and use that instead? I'm using the v120_xp Platform
Toolset to (re)build the solution.

—
You are receiving this because you were mentioned.

Reply to this email directly, view it on GitHub
<https://github.com/openitg/openitg/issues/109#issuecomment-313590052>, or mute
the thread

**Comment by Plut0nio** [comment]
@chrispable Alright, I tried to install the DirectX 9 SDK twice (I believe this is it, please correct me if I'm wrong: https://www.microsoft.com/en-us/download/details.aspx?id=6812) and I got Error S1023 both times. I even uninstalled Visual C++ before my second attempt. Should I try installing the SDK on a clean VM without Visual Studio installed? And afterwards should I try installing and using VS2010 instead? I want to try all this as soon as I get home from work. Thanks.

_Sent from my Samsung SM-G900P using [FastHub](https://play.google.com/store/apps/details?id=com.fastaccess.github)_

**Comment by chrispable** [comment]
Try this http://filehippo.com/download_directx_sdk/

Mind you, you need the 32bit version of the sdk. Don't do 2010, that
version is a paid version of msvc. Stick with 2013 or 2015. You also, of
installed in a weird place, need to include the dxsdk in your project file.
You can set the DXSDK environment variable to hold its location so the
project picks it up.

**Comment by Plut0nio** [comment]
Alright, it appears to have worked (OpenITG-SSE2.exe was generated), but when I open the .exe (on the VM), it says that it can't open because avcodec-57.dll is missing from my computer. It appears that I need to install ffmpeg to fix this (correct me if I'm wrong), but I was wondering if there was anything else that I needed to do before I transfer the whole program over to my XP machine using a flash drive.

**Comment by chrispable** [comment]
You can't run it from the compile structure. You must run the script to
create the installer. That will make an inst-temp folder that everything
can run from.

**Comment by Plut0nio** [comment]
Crap, I've only been following the instructions on README.md and not README-Windows-build.md as well. I just tried the first step of the latter using Developer Command Prompt for VS2013 (running as admin) which I found in a folder called "Visual Studio Tools." I got a message saying "Could Not Find c:\Program Files\Microsoft Visual Studio 12.0\VC\bin\lpng1256\libpng.lib" when I ran nmake. Is this an issue, or should I move on and install Msys/mingw-w32, ffmpeg, and NSIS? Also, do I need the specific versions of lpng, ffmpeg, and NSIS listed in README-Windows-build.md (1.2.56, 3.0.3, and 2.x, respectively), or should I get the newest versions of all those instead?

Sorry if I'm being overly specific or if I seem like a complete idiot. I've never really built a program like this before, I'm used to just downloading either installers in the form of .exes or the entire installed program in a compressed folder. I just really want this to work and I don't want to mess anything up.

**Comment by chrispable** [comment]
The dependencies are done, from the readme:
Windows Dependancy Build Instructions

Note for VS 2010, 2013, and 2015 these are already made for you. Nothing
more need be done.

All you need is:
NSIS Installer 2.x (https://nsis.sourceforge.net)


Then follow step 5:
To build the installer, from cmd or powershell

    assets\win32-installer\setup-installer-dir.bat
    cd inst-tmp
    makensis /nocd ..\assets\win32-installer\OpenITG.nsi

Perhaps it should be more clear that nsis isn't a dependancy.

**Comment by Plut0nio** [comment]
Alright, the installer .exe appears to have been successfully generated! Is that file the only thing I need to install or are there other files as well that I have to transfer to the XP machine?

**Comment by chrispable** [comment]
Depending on your link settings you may need the proper crt installed on
your xp machine. But in general, that's all that's needed. Sounds like your
visual studio setup on your regular pc is fubar though. You'll likely be
relegated to virtual machines unless you reinstall windows to fix it. You
can probably fix it without a reinstall but with how inexperienced in the
build chain you are your probably better off starting fresh. I ask you mark
this as not a bug and close it because there's no issue with the project.

**Comment by Plut0nio** [comment]
I just tried installing on the XP machine using just the OpenITG-setup.exe file. I got the same "avcodec-57.dll missing" error that I got earlier when I opened upb the resulting OpenITG-SSE2.exe. Also, the desktop shortcuts created point to OpenITG-PC.exe and OpenITG-PC-SSE2.exe (neither files exist, at least not in the Program directory).

Should I ask about this somewhere else? I understand that it's an issue with something _I'm_  doing and not an issue with the project itself, I just want to know what I'm doing wrong.

_Sent from my Samsung SM-G900P using [FastHub](https://play.google.com/store/apps/details?id=com.fastaccess.github)_

**Comment by chrispable** [comment]
Check your compilation directory structure for those DLLs and make sure
they are in the same folder as the exe.

On Jul 8, 2017 11:04 AM, "Plut0nio" <notifications@github.com> wrote:

I just tried installing on the XP machine using just the OpenITG-setup.exe
file. I got the same "avcodec-57.dll missing" error that I got earlier when
I opened upb the resulting OpenITG-SSE2.exe. Also, the desktop shortcuts
created point to OpenITG-PC.exe and OpenITG-PC-SSE2.exe (neither files
exist, at least not in the Program directory).

Should I ask about this somewhere else? I understand that it's an issue
with something *I'm* doing and not an issue with the project itself, I just
want to know what I'm doing wrong.

*Sent from my Samsung SM-G900P using FastHub
<https://play.google.com/store/apps/details?id=com.fastaccess.github>*

—
You are receiving this because you were mentioned.

Reply to this email directly, view it on GitHub
<https://github.com/openitg/openitg/issues/109#issuecomment-313864884>, or mute
the thread

**Comment by Plut0nio** [comment]
Okay, I just found avcodec-57.dll in C:\Users\IEUser\Source\Repos\openitg\extern\ffmpeg-build\bin. Should I just copy all the .dll files from there to OpenITG\Program after installing using the setup .exe?

**Comment by chrispable** [comment]
Yes

**Comment by Plut0nio** [comment]
I got it working! No crash on Graphics Options either! I also had to install Visual C++ VS2013 and DirectX 9 (June 2010) Redistributable Packages afterwards. Now I just have to deal with timing/lag/sync issues (playing on keyboard, I have to hit notes quite earlier than I think I have to) and getting the machine to recognize my DDR pads on all USB ports (I'm 99.9% sure that this is an issue with the machine rather than the game).

I can't thank you enough for helping me through this! I guess I'll close this now.

_Sent from my Samsung SM-G900P using [FastHub](https://play.google.com/store/apps/details?id=com.fastaccess.github)_
